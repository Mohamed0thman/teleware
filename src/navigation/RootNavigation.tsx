import {StyleSheet} from 'react-native';
import React, {useEffect, useRef} from 'react';
import {
  NavigationContainer,
  NavigationContainerRef,
} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import HomeScreen from '../screens/HomeScreen';
import LanguageScreen from '../screens/LanguageScreen';
import LocationScreen from '../screens/LocationScreen';
import ChartScreen from '../screens/ChartScreen';
import NetInfo from '@react-native-community/netinfo';
import DisconnectionScreen from '../screens/DisconnectionScreen';
import {Currency} from '../models';
import {useAppSelector} from '../store/configureStore';
import SettingsScreen from '../screens/SettingsScreen';

export type RootStackParamList = {
  Home: undefined;
  Language: undefined;
  Location: undefined;
  Chart: {currency: Currency} | undefined;
  Disconnection: undefined;
  Settings: undefined;
};

const Stack = createStackNavigator<RootStackParamList>();

const RootNavigation = () => {
  const navigationRef =
    useRef<NavigationContainerRef<RootStackParamList>>(null);

  const {lang} = useAppSelector(state => state.setting);

  useEffect(() => {
    const unsubscribe = NetInfo.addEventListener(state => {
      if (!state.isConnected) {
        navigationRef.current?.navigate('Disconnection');
      }
    });

    return () => unsubscribe();
  }, []);

  return (
    <NavigationContainer ref={navigationRef}>
      <Stack.Navigator
        initialRouteName="Language"
        screenOptions={{
          headerShown: false,
        }}>
        {!lang && <Stack.Screen name="Language" component={LanguageScreen} />}
        <Stack.Screen name="Home" component={HomeScreen} />
        <Stack.Screen name="Location" component={LocationScreen} />
        <Stack.Screen
          options={{headerShown: true}}
          name="Chart"
          component={ChartScreen}
        />
        <Stack.Screen name="Disconnection" component={DisconnectionScreen} />
        <Stack.Screen
          options={{headerShown: true}}
          name="Settings"
          component={SettingsScreen}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default RootNavigation;

const styles = StyleSheet.create({});
