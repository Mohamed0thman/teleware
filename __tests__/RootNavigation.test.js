import * as React from 'react';
import {render, screen, fireEvent} from '@testing-library/react-native';
import RootNavigation from '../src/navigation/RootNavigation';
import {renderWithRedux} from '../__helper__';

describe('Testing react navigation', () => {
  it('renders correctly', () => {
    renderWithRedux(<RootNavigation />);
  });

  it('snapshot', () => {
    expect(renderWithRedux(<RootNavigation />)).toMatchSnapshot();
  });

  it('page header', async () => {
    const {findByTestId} = renderWithRedux(<RootNavigation />);
    const header = await findByTestId('screen-title');
    expect(header).toBeDefined();
  });
});
