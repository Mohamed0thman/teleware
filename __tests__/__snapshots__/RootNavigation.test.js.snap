// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing react navigation snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          gestureResponseDistance={
            {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          pointerEvents="box-none"
          sheetAllowedDetents="large"
          sheetCornerRadius={-1}
          sheetExpandsWhenScrolledToEdge={true}
          sheetGrabberVisible={false}
          sheetLargestUndimmedDetent="all"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              [
                {
                  "display": "flex",
                  "overflow": undefined,
                },
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              {
                "close": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                handlerTag={2}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  {
                    "flex": 1,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    [
                      {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      [
                        {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          [
                            {
                              "paddingBottom": 0,
                              "paddingTop": 0,
                            },
                            {
                              "backgroundColor": "#FFFFFF",
                              "flex": 1,
                            },
                            {},
                          ]
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "color": "#000",
                              },
                              {
                                "textAlign": "left",
                              },
                              {
                                "color": "#000",
                                "fontFamily": "Roboto-Bold",
                                "fontSize": 22,
                                "marginTop": 65.5,
                                "textAlign": "center",
                              },
                            ]
                          }
                          testID="screen-title"
                        >
                          app:titles.langScreen
                        </Text>
                        <View
                          style={
                            [
                              {
                                "alignItems": "center",
                                "flex": 1,
                                "flexDirection": "row",
                                "gap": 50,
                                "justifyContent": "center",
                              },
                              {
                                "flexDirection": "row",
                              },
                            ]
                          }
                        >
                          <View
                            accessibilityState={
                              {
                                "busy": undefined,
                                "checked": undefined,
                                "disabled": undefined,
                                "expanded": undefined,
                                "selected": undefined,
                              }
                            }
                            accessibilityValue={
                              {
                                "max": undefined,
                                "min": undefined,
                                "now": undefined,
                                "text": undefined,
                              }
                            }
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              {
                                "alignItems": "center",
                                "borderColor": "#000",
                                "borderWidth": 1,
                                "height": 197,
                                "justifyContent": "center",
                                "opacity": 1,
                                "width": 160,
                              }
                            }
                            testID="en-btn"
                          >
                            <SvgMock
                              height={82}
                              width={100}
                            />
                            <Text
                              style={
                                {
                                  "color": "#000",
                                  "fontFamily": "Roboto-Bold",
                                  "fontSize": 20,
                                }
                              }
                            >
                              English
                            </Text>
                          </View>
                          <View
                            accessibilityState={
                              {
                                "busy": undefined,
                                "checked": undefined,
                                "disabled": undefined,
                                "expanded": undefined,
                                "selected": undefined,
                              }
                            }
                            accessibilityValue={
                              {
                                "max": undefined,
                                "min": undefined,
                                "now": undefined,
                                "text": undefined,
                              }
                            }
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              {
                                "alignItems": "center",
                                "borderColor": "#000",
                                "borderWidth": 1,
                                "height": 197,
                                "justifyContent": "center",
                                "opacity": 1,
                                "width": 160,
                              }
                            }
                            testID="ar-btn"
                          >
                            <SvgMock
                              height={82}
                              width={100}
                            />
                            <Text
                              style={
                                {
                                  "color": "#000",
                                  "fontFamily": "Roboto-Bold",
                                  "fontSize": 20,
                                }
                              }
                            >
                              العربيه
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;
